core-tests/shared/src/test/scala/zio/FiberSpec.scala

```diff
@@ -162,7 +162,38 @@ object FiberSpec extends ZIOBaseSpec {
             currentFiber <- ZIO.succeed(Fiber.currentFiber())
           } yield assertTrue(currentFiber.isDefined)
         }
-      } @@ TestAspect.fromLayer(Runtime.enableCurrentFiber)
+      } @@ TestAspect.fromLayer(Runtime.enableCurrentFiber),
+      suite("FiberFailure stack trace handling")(
+        test("FiberFailure captures the full stack trace including user code") {
+          def subcall(): Unit =
+            Unsafe.unsafe { implicit unsafe =>
+              Runtime.default.unsafe.run(ZIO.fail("boom")).getOrThrowFiberFailure()
+            }
+          def call1(): Unit = subcall()
+
+          val fiberFailureTest = ZIO
+            .attempt(call1())
+            .catchAll {
+              case fiberFailure: FiberFailure =>
+                val stackTrace = fiberFailure.getStackTrace.mkString("\n")
+                ZIO.log(s"Captured Stack Trace:\n$stackTrace") *>
+                  ZIO.succeed(stackTrace)
+              case other =>
+                ZIO.succeed(s"Unexpected failure: ${other.getMessage}")
+            }
+            .asInstanceOf[ZIO[Any, Nothing, String]]
+
+          fiberFailureTest.flatMap { stackTrace =>
+            ZIO.succeed {
+              assertTrue(
+                stackTrace.contains("call1") &&
+                  stackTrace.contains("subcall") &&
+                  stackTrace.contains("FiberSpec")
+              )
```

Comment:
I don't think this test tells us much. It doesn't test whether the final stack trace is correct, only that these methods appear in it. I'd like for this test to explicitly test the entire stack trace (no need to check the line numbers since those can change).

We also need to test that the `getStackTrace`, `toString` and `printStackTrace` are consistent.

Reply:
> I don't think this test tells us much. It doesn't test whether the final stack trace is correct, only that these methods appear in it. I'd like for this test to explicitly test the entire stack trace (no need to check the line numbers since those can change).

How can we expect the entire expected trace explicity? IMO it would be good let test with the some of imp methods that we can expect something like above added.

This are logs get captured:
```
zio.FiberSpec.spec.subcall(FiberSpec.scala:170)
java.base/java.lang.Thread.getStackTrace(Thread.java:2450)
zio.FiberFailure.<init>(FiberFailure.scala:31)
zio.FiberFailure$.apply(FiberFailure.scala:61)
zio.Exit.$anonfun$getOrThrowFiberFailure$1(ZIO.scala:6452)
zio.Exit.getOrElse(ZIO.scala:6445)
zio.Exit.getOrElse$(ZIO.scala:6443)
zio.Exit$Failure.getOrElse(ZIO.scala:6650)
zio.Exit.getOrThrowFiberFailure(ZIO.scala:6452)
zio.Exit.getOrThrowFiberFailure$(ZIO.scala:6451)
zio.Exit$Failure.getOrThrowFiberFailure(ZIO.scala:6650)
zio.FiberSpec$.$anonfun$spec$134(FiberSpec.scala:170)
zio.Unsafe$.unsafe(Unsafe.scala:37)
zio.FiberSpec$.subcall$1(FiberSpec.scala:169)
zio.FiberSpec$.call$1(FiberSpec.scala:172)
zio.FiberSpec$.$anonfun$spec$136(FiberSpec.scala:175)
scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
zio.ZIOCompanionVersionSpecific.$anonfun$attempt$1(ZIOCompanionVersionSpecific.scala:100)
zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:989)
zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1058)
zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1089)
zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1058)
zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1089)
zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1153)
zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1089)
zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1153)
zio.internal.FiberRuntime.evaluateEffect(FiberRuntime.scala:410)
zio.internal.FiberRuntime.evaluateMessageWhileSuspended(FiberRuntime.scala:486)
zio.internal.FiberRuntime.drainQueueOnCurrentThread(FiberRuntime.scala:246)
zio.internal.FiberRuntime.run(FiberRuntime.scala:136)
zio.internal.ZScheduler$$anon$3.run(ZScheduler.scala:380)
```

> We also need to test that the getStackTrace, toString and printStackTrace methods are consistent.

Done

Reply:
@asr2003 the stack trace you provided there is wrong. We're supposedly not including internal ZIO methods but it's clearly printed there. This is why I want the tests to test the stack trace explicitly

Reply:
@kyri-petrou We are including the both stacktraces in the combined state so we are getting this over.